### PR TITLE
Cleanup preferences

### PR DESCRIPTION
**Pull request by calixtus**
Extracted some calls to Globals.prefs and Globals.stateManager.
Step by step.

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by koppor — COMMENTED** [review]
This is a good preparation for using dependency injection

**Review by koppor — APPROVED** [review]

File: src/main/java/org/jabref/gui/slr/ExistingStudySearchAction.java [hunk 1]
```diff
@@ -117,11 +125,8 @@ protected void setupRepository(Path studyRepositoryRoot) throws IOException, Git
...
     private Path getInitialDirectory(Path standardWorkingDirectory) {
-        if (frame.getBasePanelCount() == 0) {
-            return standardWorkingDirectory;
-        } else {
-            Optional<Path> databasePath = frame.getCurrentLibraryTab().getBibDatabaseContext().getDatabasePath();
-            return databasePath.map(Path::getParent).orElse(standardWorkingDirectory);
-        }
+        return stateManager.getActiveDatabase().map(database -> database.getDatabasePath().map(Path::getParent)
+                                                                        .orElse(standardWorkingDirectory))
+                           .orElse(standardWorkingDirectory);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/slr/ExistingStudySearchAction.java:130]: Doppelt gemoppelt?

Comment (calixtus) [reply src/main/java/org/jabref/gui/slr/ExistingStudySearchAction.java:130]: getParent is different in the first. Took it from the original by Dom

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/slr/ExistingStudySearchAction.java:130]: But the orElse is duplicated. The inner one can be removed

File: src/main/java/org/jabref/gui/slr/ExistingStudySearchAction.java [hunk 2]
```diff
@@ -38,19 +39,26 @@
...
+                                     StateManager stateManager) {
         this.frame = frame;
         this.dialogService = frame.getDialogService();
         this.fileUpdateMonitor = fileUpdateMonitor;
-        this.workingDirectory = getInitialDirectory(preferencesService.getWorkingDir());
         this.taskExecutor = taskExecutor;
         this.preferencesService = preferencesService;
-        this.importFormatPreferneces = preferencesService.getImportFormatPreferences();
+        this.stateManager = stateManager;
+        this.importFormatPreferences = preferencesService.getImportFormatPreferences();
```
Comment (btut) [inline src/main/java/org/jabref/gui/slr/ExistingStudySearchAction.java:58]: Why keep both preferencesService and the smaller preferences objects?

Comment (calixtus) [reply src/main/java/org/jabref/gui/slr/ExistingStudySearchAction.java:58]: As a temporary placeholder on the way to use only the small objects.

**Review by Siedlerchr — APPROVED** [review]

**Review by btut — APPROVED** [review]
LGTM

**Pull request merged**